### ISSUE TITLE
Microsoft Office 365 as authentication source

### ISSUE THREAD

**Issue by michzimny** [up2university/single-sign-on#49]
This is an improvement suggestion collected from external users. Do you have any idea how problematic it could be, and how realistic it is to have it?

**Comment by foobarable** [comment]
We have yet to look at the authentication protocol they use.

**Comment by michzimny** [comment]
@foobarable, would you be able to provide, till the next week, a description for a few sentences explaining what could be done on this matter? We're preparing a report on improvement suggestions from users and we need to address this issue anyhow.

**Comment by foobarable** [comment]
Currently reading about it. It looks like we would need an AD for doing so.

**Comment by michzimny** [comment]
And AD stands for? :)

**Comment by foobarable** [comment]
Active Directory. But I've found some documentation about Azure Directory as OpenIDC provider. Let me check it.

**Comment by foobarable** [comment]
It's not possible for me to register with my work email at microsoft to do this:

https://docs.microsoft.com/de-de/azure/active-directory/develop/v1-protocols-openid-connect-code

**Comment by michzimny** [comment]
So shall we create a private email account to be used just for the purposes of AD? And have the password for this account shared with someone else just to not loose the access?

**Comment by foobarable** [comment]
I created an account but I'm not sure if it will help us. I now have my own Azure AD

**Comment by michzimny** [comment]
I think it's rather usual in such cases that a service administrator (you as the admin of the SSO) creates such account at IdP to enable the service (our SSO) to work with that IdP...

**Comment by foobarable** [comment]
Yes, but using an empty AD doesn't help us any bit. Looks like I need some reading.

**Comment by foobarable** [comment]
Okay, I think we would have to integrate _each_ Office 365 instance to be able for them to authenticate. 
They would need to create the AD OpenIDC application in their portal. It's one thing to integrate those authentication sources technically, but it will be not so easy from UI perspective.

**Comment by michzimny** [comment]
My colleagues have also given me the following link. Maybe it helps a bit.
https://stackoverflow.com/questions/40088260/create-application-with-authenticates-against-o365-azure-ad-with-openidconnect

**Comment by michzimny** [comment]
I was told that Ms Office 365 accounts can be self-hosted somehow (via the AD?) or used in SaaS model. In the former case, admins have also an option to export local user accoutns to the central instance (https://docs.microsoft.com/en-us/office365/enterprise/office-365-integration).

So, I think we can ignore local AD, and just enable authentication against central Office 365. Then, if local admins permit, any Office 365 user will be able to authenticate this way to Up2U.

**Comment by michzimny** [comment]
@foobarable, would you be able to investigate feasibility of that approach mentioned in my last comment?
